File: PRAKTIKUM I/Modul 6/Python/PRAK604-2310817110015-AKHMAD_CHAIDAR_ANANDA.py
# mendefinisikan total string dan maksimal string
TOTAL_STRINGS = 2
MAX_STRINGS = 50

# deklarasi variabel
pesan = ["", ""]
hitung_bintang = 0
hitung_pagar = 0
ubah_pesan = [""]
pesan_baru = [""]

# input
for i in range(TOTAL_STRINGS):
    pesan[i] = input()

# hitung panjang dari kedua pesan
# dan mengkonversikannya ke tipe data integer
size1 = int(len(pesan[0]))
size2 = int(len(pesan[1]))

# kondisi dan perulangan
if size1 != size2:
    print("Panjang kalimat berbeda, pesan palsu")
elif size1 == size2:
    for i in range(size1 and size2):
        if pesan[0][i] == ' ' and pesan[1][i] == ' ':
            pesan_baru.append(' ')
            #ubah_pesan[i] == ' '
        elif pesan[0][i] == pesan[1][i]:
            pesan_baru.append('*')
            #ubah_pesan[i] == '*'
            hitung_bintang += 1
        else:
            pesan_baru.append('#')
            #ubah_pesan[i] == '#'
            hitung_pagar += 1

    # menggabungkan list pesan baru menjadi string
    print(''.join(pesan_baru))
    print(f"* = {hitung_bintang}")
    print(f"# = {hitung_pagar}")

    if hitung_bintang >= hitung_pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

File: PRAKTIKUM I/Modul 6/C/PRAK604-2310817110015-AKHMAD_CHAIDAR_ANANDA.c
#include<stdio.h>
#include<string.h>

#define TOTAL_STRINGS 2
#define MAX_STRINGS 50

int main(){
    // deklarasi variabel 
    char pesan[TOTAL_STRINGS][MAX_STRINGS];
    int hitung_bintang = 0, hitung_pagar = 0;
    char ubah_pesan[MAX_STRINGS];

    // deklarasi variabel perulangan
    int i, j;
    // input kedua pesan
    for (i = 0; i < TOTAL_STRINGS; i++){
        scanf("%[^\n]%*c", &pesan[i]);
    }

    // menghitung panjang dari kedua pesan
    int size1 = strlen(pesan[0]);
    int size2 = strlen(pesan[1]);

    // kondisi
    if (size1 != size2){
        printf("Panjang kalimat berbeda, pesan palsu"); // kondisi jika panjang pesan 1 tidak sama dengan pesan 2
    } else if (size1 == size2) { // kondisi jika panjang pesan 1 sama dengan pesan 2
        for (i = 0; i < size1; i++){
            if (pesan[0][i] == ' ' && pesan[1][i] == ' ') {
            ubah_pesan[i] = ' '; // Jika keduanya spasi akan menambahkan spasi pada output
            } else if (pesan[0][i] == pesan[1][i]) {
                ubah_pesan[i] = '*';
                hitung_bintang++;
            } else {
                ubah_pesan[i] = '#';
                hitung_pagar++;
            }
        }
        // menambahkan karakter kosong(null)
        // pada akhir baris string dalam array pada output
        ubah_pesan[size1] = '\0'; 

        // output 
        printf("%s\n", ubah_pesan);
        printf("* = %d\n", hitung_bintang);
        printf("# = %d\n", hitung_pagar);

        // kondisi apakah pesan asli atau palsu
        if (hitung_bintang >= hitung_pagar){
            printf("Pesan Asli\n");
        } else printf("Pesan Palsu\n");
    }

    return 0;
}